pkg/inventory/inventory-client.go

```diff
@@ -344,10 +335,17 @@ func (cic *ClusterClient) getClusterInventoryObjsByName(inv Info) (object.Unstru
 	if apierrors.IsNotFound(err) {
 		return object.UnstructuredSet{}, nil
 	}
+	if inv.ID() != "" {
+		if inventoryID, err := retrieveInventoryLabel(clusterInv); err != nil {
+			return nil, err
+		} else if inv.ID() != inventoryID {
+			return nil, fmt.Errorf("inventory-id of inventory object in cluster doesn't match provided id %q", inv.ID())
```

Comment:
error should probably include the name & namespace of the problematic inventory object, not just the id

Reply:
Done